docker-app/qfieldcloud/core/views/projects_views.py

```diff
@@ -101,18 +101,24 @@ class ProjectViewSet(viewsets.ModelViewSet):
     permission_classes = [permissions.IsAuthenticated, ProjectViewSetPermissions]
 
     def get_queryset(self):
-        include_public = False
-        include_public_param = self.request.query_params.get(
-            "include-public", default=None
-        )
-        if include_public_param and include_public_param.lower() == "true":
-            include_public = True
 
         projects = Project.objects.for_user(self.request.user)
-        if not include_public:
-            projects = projects.exclude(
-                user_role_origin=ProjectQueryset.RoleOrigins.PUBLIC
+
+        # In the list endpoint, by default we filter out public projects. They can be
+        # included with the `include-public` query parameter.
+        if self.action == "list":
```

Comment:
I think this should fix the error 500 in test_project.py

(the queryset filtered out public projects, which makes sense in the list endpoint, but not in the get endpoint)

@suricactus Can you check this fix makes sense ? Did the get endpoint work for public projects so far ?

Reply:
Ok, sounds good.